qt/component-box.cpp

```diff
@@ -30,7 +30,7 @@ class AppStream::ComponentBoxData : public QSharedData
 public:
     ComponentBoxData(ComponentBox::Flags flags)
     {
-        m_cbox = as_component_box_new(static_cast<AsComponentBoxFlags>(flags.toInt()));
+        m_cbox = as_component_box_new(static_cast<AsComponentBoxFlags>(flags==int()));
```

Comment:
I have never seen that `flags==int()` syntax - aren't we converting a nonsense bool to `AsComponentBoxFlags` here with that?

AFAIK casting QFlags to int should also have the same effect as explicitly calling toInt, so maybe that's a solution (but if there's some magic going on here that I don't know, can you explain it? I searched the Qt documentation and didn't find anything mentioning that this works).